### ISSUE TITLE
Dashboard complaining about old bokeh version

### ISSUE THREAD

**Issue by m-albert** [dask/distributed#4798]
Hi, on my system the `dask` dashboard is not accessible. Thanks a lot for any hints!

**What happened**:

Browser displays (despite bokeh version 2.3.1 being installed):
>Dask needs bokeh >= 0.13.0 for the dashboard.
Install with conda: conda install bokeh>=0.13.0
Install with pip: pip install bokeh>=0.13.0

This is with a fresh conda environment with only `bokeh` and `distributed` installed using pip.

**What you expected to happen**:

The dashboard to show normally, or at least not to have problems with the `bokeh` version, which seems fine.

**Minimal Complete Verifiable Example**:

```python
from dask.distributed import Client
client = Client()
```

**Environment**:

- Dask version: 2021.04.1
- Python version: 3.9.4
- Operating System: macOS 10.14.6
- Install method (conda, pip, source): pip

**Comment by fjetter** [comment]
Can you execute `import distributed.dashboard.scheduler` in a fresh python interpreter and tell us what you see? The "missing bokeh" message is coupled to this import and I suspect something else is going on.
I couldn't reproduce with neither conda not pip environment.

If that doesn't help to resolve the issue, can you please provide a full `pip freeze`? Also, do you have an intel or M1 mac book (If you don't know, that's likely an intel but you can find out with the "Apple -> About This Mac" dialog under "Chip")?

**Comment by m-albert** [comment]
Thanks @fjetter :)

> Can you execute import distributed.dashboard.scheduler in a fresh python interpreter and tell us what you see? The "missing bokeh" message is coupled to this import and I suspect something else is going on.

The import works fine and doesn't print or log anything. How else can the "missing bokeh" message be triggered?

> If that doesn't help to resolve the issue, can you please provide a full pip freeze?

Here's my `pip freeze`:

bokeh==2.3.1
certifi==2020.12.5
click==7.1.2
cloudpickle==1.6.0
dask==2021.4.1
distributed==2021.4.1
fsspec==2021.4.0
HeapDict==1.0.1
Jinja2==2.11.3
locket==0.2.1
MarkupSafe==1.1.1
msgpack==1.0.2
numpy==1.20.2
packaging==20.9
partd==1.2.0
Pillow==8.2.0
psutil==5.8.0
pyparsing==2.4.7
python-dateutil==2.8.1
PyYAML==5.4.1
six==1.16.0
sortedcontainers==2.3.0
tblib==1.7.0
toolz==0.11.1
tornado==6.1
typing-extensions==3.10.0.0
zict==2.0.0

>Also, do you have an intel or M1 mac book

It's an intel chip.

Thanks so much for your help!

**Comment by fjetter** [comment]
> How else can the "missing bokeh" message be triggered?

I am not aware of any other way

The relevant code sections which arms this warning can be found here https://github.com/dask/distributed/blob/a2a0feb33b19eea27f0ff73fe12acfa8a90d73b6/distributed/scheduler.py#L3332-L3345

TLDR if the above mentioned import cannot be found, the status page / dashboard will be replaced with the warning you're seeing.

---

Even with the above `pip freeze`, I cannot reproduce. I am running python `3.9.2`, not `3.9.4` but that should hopefully not matter. I also had to install numpy and pillow using conda since the pip wheels for my platform where not available. Still, that should not make a difference.

Have you tried recreating the environment you are running in? You mentioned that you are using conda to setup the environment. Is there a reason why you are not installing the dependencies from there? Does it make a difference if you install all deps using conda?

**Comment by m-albert** [comment]
> However, if you say you can reproduce the issue with the above two lines, the import ordering should be the same for you and me.

Yeah, it's really just the above two lines. Also starting a `dask-scheduler` from the command line leads to the same error page.

>Have you tried recreating the environment you are running in? You mentioned that you are using conda to setup the environment. Is there a reason why you are not installing the dependencies from there? Does it make a difference if you install all deps using conda?

Also, if I install the same minimal environment (just `bokeh` and `distributed`) using `conda` the same happens...

**Comment by m-albert** [comment]
Interestingly, I found an old environment in which the dashboard still works. I recreated it in a new environment with

bokeh==1.0.4
dask==2.0.0
distributed==2.0.1

and here the dashboard also works.

**Comment by fjetter** [comment]
> Interestingly, I found an old environment in which the dashboard still works.

That's not old, that's ancient ;)

> Also, if I install the same minimal environment (just bokeh and distributed) using conda the same happens...

I tried at least four different ways to install the environment on my machine but couldn't reproduce this, yet. J

* I occasionally had an issue with the `zsh` integration of conda, such that a conda activate would not reliably work and would still select the wrong interpreter. You can verify this with `which python` and ensure this points to the interpreter you _think_ it should be. That should print something like `/path/to/conda/envs/<environment_name>/bin/python`. If this is wrong, that's likely an explanation. Starting a fresh shell helps when I hit this issue. Same goes for things like `which dask-scheduler` which should point to the respective bin directory
* If everything checks out with the shell and environment, the only thing I could otherwise come up with is to bisect and find the culprit (https://git-scm.com/docs/git-bisect).

**Comment by m-albert** [comment]
>That's not old, that's ancient ;)

Indeed `dask` did change quite a bit since then 😅

The interpreter seems fine. Also I placed a `pdb.set_trace()` after line 3341 in the code you posted, and verified that `missing_bokeh==False` as the client is instantiated.

**Comment by m-albert** [comment]
Hmm okay, so some dashboard pages such as

http://localhost:8787/tasks
http://localhost:8787/system

do work. Basically everything apart form status.
